src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -55,16 +91,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   // Send a user to datastore. Update the current information about the user if one exists.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGitHubID = request.getParameter("githubID");
-    if (userGitHubID.trim().isEmpty()) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+    String userGitHubAuthToken;
+    ArrayList<String> userTags;
+    // Get information from request body
+    try {
+      userGitHubAuthToken = request.getParameter("gitHubAuthToken");
+      userTags = new ArrayList<>(Arrays.asList(request.getParameterValues("interestTags")));
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace() + "/n/n Incomplete POST body parameters",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "User information was incomplete");
+      return;
+    }
+    // Get User information from GitHub using the Oath token.
+    GHMyself userGHMyself;
+    try {
+      GitHub userPersonalGitHubInstance = GitHub.connectUsingOAuth(userGitHubAuthToken);
+      userGHMyself = userPersonalGitHubInstance.getMyself();
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage()
+              + "/n/n"
+              + e.getStackTrace()
+              + "/n/n Could not get Authenticate User with the given token",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "User could not be authenticated by GitHub, please try again");
+      return;
+    }
+    // Build and send the User's datastore entity
+    try {
+      Entity personEntity =
+          new PersonBuilder()
+              .userID(AuthServlet.getAuthorizedUser().getUserId())
+              .gitHubID(userGHMyself.getLogin())
+              .email(userGHMyself.getEmail())
+              .interestTags(userTags)
+              .buildPersonEntity();
+      datastore.put(personEntity);
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage()
+              + "/n/n"
+              + e.getStackTrace()
+              + "/n/n User informatiuon could not be sent to datastore",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "User information could not be sent to datastore");
       return;
     }
```

Comment:
I think this is another instance where the error being returned to the user is "unactionable," and I think it also merits an error code in the 5xx region instead of the error code for bad request. When do we anticipate encountering an error here, and can we more specifically handle that error with a response?